packages/clay-tabs/src/Item.tsx

```diff
@@ -0,0 +1,185 @@
+	/**
+	 * Flag to indicate if the component is active or not.
+	 */
+	active?: boolean;
+
+	/**
+	 * Tab Item component to render. Can be an 'anchor' or a 'button'.
+	 */
+	component?: ElementType;
+
+	/**
+	 * Flag to indicate if the TabPane is disabled.
+	 */
+	disabled?: boolean;
+
+	/**
+	 * Flag to indicate if the Tab Item is a dropdown.
+	 */
+	dropdown?: boolean;
+
+	forwardRef?: React.Ref<any>;
+
+	/**
+	 * Props to be added to the item element that can be an anchor or a button.
+	 */
+	itemElementProps?: React.HTMLAttributes<
+		HTMLAnchorElement | HTMLButtonElement
+	>;
+
+	/**
+	 * Registers the `tabkey` for the Tab Item.
+	 */
+	tabkey: number;
+
+	/**
+	 * Callback to be used when clicking to a Tab Item.
+	 */
+	onClick?: (val: number) => void;
+
+	/**
+	 * The path to the SVG spritemap file containing the icons.
+	 */
+	spritemap?: string;
+
+	/**
+	 * Name of the Tab
+	 */
+	tabName: string;
+}
+
+const TabHeader = React.forwardRef(
+	(
+		{
+			active = false,
+			component = 'button',
+			disabled = false,
+			dropdown = false,
+			forwardRef,
+			itemElementProps = {},
+			tabkey,
+			onClick,
+			spritemap,
+			tabName,
+		}: IProps,
+		IProps
+	) => {
+		if (component === 'anchor') {
+			return (
+				<>
+					<a
+						aria-controls={tabName.toLowerCase()}
+						aria-disabled={!active}
+						aria-selected={!active}
+						className={classNames('nav-link', {
+							active,
+							disabled,
+						})}
+						data-testid={`${tabName.trim().toLowerCase()}TabItem`}
+						href={`#${tabName.trim().toLowerCase()}`}
+						id={`${tabName}Tab`}
+						onClick={() => onClick && onClick(tabkey)}
+						ref={forwardRef}
+						role="tab"
+						tabIndex={disabled ? -1 : undefined}
+						{...itemElementProps}
+					>
+						{tabName}
+						{dropdown && (
+							<ClayIcon
+								spritemap={spritemap}
+								symbol="caret-bottom"
+							/>
+						)}
+					</a>
+				</>
+			);
+		}
+
+		return (
+			<>
+				<button
+					aria-controls={tabName.toLowerCase()}
+					aria-disabled={!active}
+					aria-selected={active}
+					className={classNames('btn btn-unstyled nav-link', {
+						active,
+						disabled,
+					})}
+					data-testid={`${tabName.trim().toLowerCase()}TabItem`}
+					id={`${tabName}Tab`}
+					onClick={() => onClick && onClick(tabkey)}
+					ref={forwardRef}
+					role="tab"
+					tabIndex={disabled ? -1 : undefined}
+					type="button"
+					{...itemElementProps}
+				>
+					{tabName}
+					{dropdown && (
+						<ClayIcon spritemap={spritemap} symbol="caret-bottom" />
+					)}
+				</button>
+			</>
+		);
+	}
+);
+
+const Item: React.FunctionComponent<IProps> = ({
+	active = false,
+	component,
+	className,
+	disabled,
+	dropdown,
+	forwardRef,
+	itemElementProps,
+	tabkey,
+	onClick = () => {},
+	spritemap,
+	tabName,
+	...otherProps
+}: IProps) => {
+	return (
+		<li
+			className={classNames(
+				'nav-item',
+				{
+					dropdown,
+				},
+				className
+			)}
+			{...otherProps}
+		>
+			{tabName && (
```

Comment:
`tabName` is a required parameter, so why the guard here?

Reply:
Yes, It's unnecessary. I was relying on Typescript to warn me about this. Removing 😅